SPECS/cloud-init/network_renaming.patch

```diff
@@ -0,0 +1,21 @@
+diff -ruN a/cloudinit/net/netops/iproute2.py b/cloudinit/net/netops/iproute2.py
+--- a/cloudinit/net/netops/iproute2.py	2024-09-04 12:01:58.000000000 -0700
++++ b/cloudinit/net/netops/iproute2.py	2024-12-11 17:00:35.590222604 -0800
+@@ -2,7 +2,7 @@
+ 
+ from cloudinit import subp
+ from cloudinit.net.netops import NetOps
+-
++import time
+ 
+ class Iproute2(NetOps):
+     @staticmethod
+@@ -30,6 +30,7 @@
+     @staticmethod
+     def link_rename(current_name: str, new_name: str):
+         subp.subp(["ip", "link", "set", current_name, "name", new_name])
++        time.sleep(0.5)
```

Comment:
Is this an upstream-provided fix? I'm concerned seeing a sleep call get added to address what is being labeled a race condition. It tends to sweep the true root cause under the rug, and often runs the risk of still not mitigating the issue appropriately for some percentage of scenarios.